### PR TITLE
Fixed .syntax on ifs with an inner if without an else

### LINKED ISSUE #3128: Syntax function does not return the correct value for Ifs with an inner if without an else
Using:
  - Scalameta version 4.7.7
  - Scala 2.13.10

Steps to reproduce:
```scala
import scala.meta._

val expr1 = q"if (someCondition1) println(123)"
val expr2 = q"if (someCondition2) ($expr1) else println(321)"

println(expr2.syntax) // this is incorrect
```

actual:
```scala
if (someCondition2) if (someCondition1) println(123) else println(321)
```

expected:
```scala
if (someCondition2) (if (someCondition1) println(123)) else println(321)

OR

if (someCondition2) if (someCondition1) println(123) else () else println(321)
```

---

The `.syntax` is changing the semantics of the expression which shouldn't happen.

### PR DESCRIPTION
**Pull request by RCMartins**
Fixes #3128

### CODE REVIEW

File: tests/shared/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala [hunk 1]
```diff
@@ -464,6 +464,8 @@ class SyntacticSuite extends scala.meta.tests.parsers.ParseSuite {
   test("some tricky parenthesization") {
     assertEquals(templStat("if (1) 2 else 3 + 4").syntax, "if (1) 2 else 3 + 4")
     assertEquals(templStat("(if (1) 2 else 3) + 4").syntax, "(if (1) 2 else 3) + 4")
+    assertEquals(templStat("if (1) (if (2) 3) else 4").syntax, "if (1) (if (2) 3) else 4")
```
Comment (kitbellew) [inline tests/shared/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala:467]: can you please split your PR into two commits, one adding the tests (and the current, pre-fix, behaviour), and the second fixing the problem and showing the difference in the tests?

File: scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala [hunk 2]
```diff
@@ -411,6 +411,11 @@ object TreeSyntax {
           case _ => block(s(), t.stats)
         }
       case t: Term.If =>
+        val needParens: Boolean =
+          guessHasElsep(t) && (t.thenp match {
+            case innerIf: Term.If => !guessHasElsep(innerIf)
+            case _ => false
+          })
```
Comment (kitbellew) [inline scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala:418]:
```suggestion
        val needParens: Boolean = t.thenp match {
          case innerIf: Term.If => !guessHasElsep(innerIf) && guessHasElsep(t)
          case _ => false
        }
```

File: scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala [hunk 3]
```diff
@@ -419,7 +424,7 @@ object TreeSyntax {
             " (",
             t.cond,
             ") ",
-            p(Expr, t.thenp),
+            if (needParens) s("(", t.thenp, ")") else p(Expr, t.thenp),
```
Comment (kitbellew) [inline scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala:427]:
```suggestion
            w("(", p(Expr, t.thenp), ")", needParens),
```

**Review by kitbellew — CHANGES_REQUESTED** [review]

File: scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala (hunk 3 above)
Comment (kitbellew) [reply scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala:427]: actually, maybe never mind, as `p()` could add own parens.

File: tests/shared/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala (hunk 1 above)
Comment (RCMartins) [reply tests/shared/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala:467]: Ok, done

File: scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala (hunk 3 above)
Comment (RCMartins) [reply scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala:426]: if `p()` had a `force: Boolean` parameter we could use `p(Expr, t.thenp, force = needParens)`, maybe that would be better?

Comment (kitbellew) [reply scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala:426]: > if `p()` had a `force: Boolean` parameter we could use `p(Expr, t.thenp, force = needParens)`, maybe that would be better?

that would be nice :)

**Review by kitbellew — APPROVED** [review]
thank you!

**Pull request merged**